src/tools/schema2base.ts

```diff
@@ -66,10 +66,32 @@ export abstract class Schema2Base {
           if (!(name in inlineSchemas)) {
             inlineSchemas[name] = descriptor.schema.model.getEntitySchema();
           }
-         }
+        }
       }
     }
 
+    return inlineSchemas;
+  }
+
+  private async processFile(src: string) {
```

Comment:
Rather than playing around with file handling in tests, consider restructuring Schema2Base for easier testing. Basically you want to separate the file I/O from the processing logic, and then just test the processing logic:

```ts
private processFile() {
  // set up file names and timestamp guard

  const generated = this.processManifest(`import ${src}`);

  const outFile = fs.openSync(outPath, 'w');
  fs.writeSync(outFile, this.fileHeader(outName));
  for (const entry of generated) {
    fs.writeSync(outFile, generated);
  }
  fs.writeSync(outFile, this.fileFooter());
  fs.closeSync(outFile);
}

processManifest(manifestStr: string) {
  const manifest = await Utils.parse(manifestStr);
  const schemas = this.collectSchemas(manifest);
  // return if none
  const inlineSchemas = this.collectInlineSchemas(schemas);
  const generated = [];
  for (const dict of [inlineSchemas, schemas]) {
    for (const [name, schema] of Object.entries(dict)) {
      generated.push(this.entityClass(name, schema).replace(/ +\n/g, '\n'));
    }
  }
  return generated;
}
```

Now you can just call `processManifest` in tests with the desired manifest string directly - and no need for static methods :).

Reply:
Good shout, will implement.